kvbc/src/categorization/block_merkle_category.cpp

```diff
@@ -157,15 +157,23 @@ std::vector<Buffer> versionedKeys(const std::vector<std::string>& keys, const st
   return versioned_keys;
 }
 
+void putLatestKeyVersion(NativeWriteBatch& batch, const Hash& key_hash, TaggedVersion version) {
+  batch.put(BLOCK_MERKLE_LATEST_KEY_VERSION_CF, key_hash, serializeThreadLocal(LatestKeyVersion{version.encode()}));
+}
+
 void putKeys(NativeWriteBatch& batch,
              uint64_t block_id,
              std::vector<KeyHash>&& hashed_added_keys,
              std::vector<KeyHash>&& hashed_deleted_keys,
              BlockMerkleInput& updates) {
   auto kv_it = updates.kv.begin();
   for (auto key_it = hashed_added_keys.begin(); key_it != hashed_added_keys.end(); key_it++) {
+    // Only serialize the Header of a DBValue, to prevent the need to copy a potentially large value.
+    auto header = serialize(DbValueHeader{false, static_cast<uint32_t>(kv_it->second.size())});
```

Comment:
Nit: I know you are not a big fan of `serializeThreadLocal()`, but I think if we do `auto header = toSlice(serializeThreadLocal(DbValueHeader{...});` we might avoid an allocation.

Reply:
Will do! I actually came around on serializeThreadLocal even more after you convinced me it was safe by reading the RocksDB code when adding support for SliceParts. I realized all the data in a batch is copied into a single vector.